src/OpenTelemetry.Instrumentation.Http/HttpClientInstrumentationOptions.cs

```diff
@@ -41,10 +41,32 @@ public class HttpClientInstrumentationOptions
         /// <remarks>
         /// <para><see cref="Activity"/>: the activity being enriched.</para>
         /// <para>string: the name of the event.</para>
-        /// <para>object: the raw object from which additional information can be extracted to enrich the activity.
+        /// <para><see cref="HttpRequestMessage"/> object from which additional information can be extracted to enrich the activity.
         /// The type of this object depends on the event, which is given by the above parameter.</para>
         /// </remarks>
-        public Action<Activity, string, object> Enrich { get; set; }
+        public Action<Activity, HttpRequestMessage> EnrichWithHttpRequestMessage { get; set; }
+
+        /// <summary>
+        /// Gets or sets an action to enrich an Activity.
```

Comment:
We should probably mention the raw object type name in the summary description as well.